packages/gatsby/src/utils/worker/__tests__/test-helpers/child-for-tests.ts

```diff
@@ -53,3 +57,50 @@ export async function runAPI(apiName: string): Promise<any> {
 export function getAPIRunResult(): string | undefined {
   return (global as any).test
 }
+
+export function getState(): CombinedState<IGatsbyState> {
+  return store.getState()
+}
+
+const runQuery = (
+  schema: GraphQLSchema,
+  schemaComposer,
+  query: string
+): Promise<ExecutionResult> =>
+  graphql(
+    schema,
+    query,
+    undefined,
+    withResolverContext({
+      schema,
+      schemaComposer,
+      context: {},
+      customContext: {},
+    })
+  )
+
+// test: schema
+export async function getRunQueryResult(): Promise<ExecutionResult> {
```

Comment:
Since the schema will be serialized between processes we need to run the queries in the worker process itself to have it working

Reply:
just a suggestion - maybe we could pass query string via argument and not hardcode it here - just so if we ever add different fixture sites we could reuse this function?